de-lib/src/main/java/org/iplantc/de/client/services/AppTemplateServices.java

```diff
@@ -34,9 +34,17 @@
      * @param appId the <code>App</code> id.
      * @param callback
      */
-    void getAppTemplateForEdit(HasId appId, DECallback<AppTemplate> callback);
+    void getAppTemplateForEdit(HasQualifiedId appId, DECallback<AppTemplate> callback);
     
     /**
+     * Retrieves a UI preview of the given <code>AppTemplate</code>.
+     * 
+     * @param at the <code>AppTemplate</code> for which the preview should be produced.
+     * @param callback
+     */
+    void getAppTemplatePreview(AppTemplate at, DECallback<AppTemplate> callback);
```

Comment:
Minor thing, but I recently removed this and its counterpart in `AppTemplateServicesImpl` as it was dead code.  It might've accidentally gotten added back in a rebase.